tests/v2/validation/provisioning/provisioning_rke1_driver_test.go

```diff
@@ -0,0 +1,185 @@
+	"github.com/stretchr/testify/suite"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+type RKE1NodeDriverProvisioningTestSuite struct {
+	suite.Suite
+	client             *rancher.Client
+	session            *session.Session
+	kubernetesVersions []string
+	cnis               []string
+	providers          []string
+}
+
+func (r *RKE1NodeDriverProvisioningTestSuite) TearDownSuite() {
+	r.session.Cleanup()
+}
+
+func (r *RKE1NodeDriverProvisioningTestSuite) SetupSuite() {
+	testSession := session.NewSession(r.T())
+	r.session = testSession
+
+	clustersConfig := new(Config)
+	config.LoadConfig(ConfigurationFileKey, clustersConfig)
+
+	r.kubernetesVersions = clustersConfig.KubernetesVersions
+	r.cnis = clustersConfig.CNIs
+	r.providers = clustersConfig.Providers
+
+	client, err := rancher.NewClient("", testSession)
+	require.NoError(r.T(), err)
+
+	r.client = client
+
+	enabled := true
+	var testuser = AppendRandomString("testuser-")
+	user := &management.User{
+		Username: testuser,
+		Password: "rancherrancher123!",
+		Name:     testuser,
+		Enabled:  &enabled,
+	}
+
+	newUser, err := users.CreateUserWithRole(client, user, "clusters-create", "user")
+	require.NoError(r.T(), err)
+
+	newUser.Password = user.Password
+}
+
+func (r *RKE1NodeDriverProvisioningTestSuite) ProvisioningRKE1Cluster(provider Provider) {
+	providerName := " Node Provider: " + provider.Name
+	tests := []struct {
+		name   string
+		client *rancher.Client
+	}{
+		{"Admin User", r.client},
+	}
+
+	var name string
+	for _, tt := range tests {
+		subSession := r.session.NewSession()
+		defer subSession.Cleanup()
+
+		client, err := tt.client.WithSession(subSession)
+		require.NoError(r.T(), err)
+
+		for _, kubeVersion := range r.kubernetesVersions {
+			name = tt.name + providerName + " Kubernetes version: " + kubeVersion
+			for _, cni := range r.cnis {
+				name += " cni: " + cni
+				r.Run(name, func() {
+					cluster := &management.Cluster{
+						DockerRootDir:           "/var/lib/docker",
+						EnableClusterAlerting:   false,
+						EnableClusterMonitoring: false,
+						LocalClusterAuthEndpoint: &management.LocalClusterAuthEndpoint{
+							Enabled: true,
+						},
+						Name:     AppendRandomString("rke1"),
+						Provider: provider.Name,
+						RancherKubernetesEngineConfig: &management.RancherKubernetesEngineConfig{
+							DNS: &management.DNSConfig{
+								Provider: "coredns",
+								Options: map[string]string{
+									"stubDomains": "cluster.local",
+								},
+							},
+							Ingress: &management.IngressConfig{
+								Provider: "nginx",
+							},
+							Monitoring: &management.MonitoringConfig{
+								Provider: "metrics-server",
+							},
+							Network: &management.NetworkConfig{
+								MTU:     0,
+								Options: map[string]string{},
+							},
+						},
+					}
+
+					clusterResp, err := client.Management.Cluster.Create(cluster)
+					require.NoError(r.T(), err)
+
+					var nodeTemplateResp *nodetemplates.NodeTemplate
+
+					switch {
```

Comment:
We don't want this switch case here. We need to create a new provider file similar to the current one, but with node template

Reply:
Thanks @igomez06. I'll look into creating a new provider file to do just this and call it within this file. Should we just call it `providers_rke1.go` or how do you think would be best to structure this?

@slickwarren mentioned in the other PR of us separating RKE1/RKE2 to avoid long-term confusion. I had some complications when separating folders out, but I do think that would be a good idea if we can make it work. What do you think?

Reply:
@slickwarren what do you think? I think having provisioning/rke1 and provisioning/rke2 makes sense for a directory structure, and having the respective providers. Or maybe for now just have a new provider_rke1.go works for me as well.

Reply:
I second having different folders for different KEs

Reply:
if you need help with how to do this, see `providers.go` which might be helpful

Reply:
I refactored into `rke1` and `rke2` and updated the `rke1/providers.go` to specifically handle nodetemplates for RKE1 provisioning; verified locally that this is working fine.

Unsure if `config.go` needs to change at all? I just copied into `rke1` because `go` prefers having all *.go files in the same directory when testing...but the initial concern of folder separation and the switch statement is resolved.

Reply:
@slickwarren what do you think about having a single `config.go` file as mentioned - would config files change depending on cluster types tests in the future? If yes, please ignore the below explanation.

If it's gonna be the same, @markusewalker, for the `config.go` files, you can use the exported things from `provisioning/config.go` instead of duplicating the file. Only the namespace name needs to be public, the rest already should be public.

Reply:
good point. I'm not sure if it would need to change in the future, but since they are the same now for rke1 and rke2, I think your approach is best. cc @markusewalker do you agree, and if so could you amend this PR again to reflect the changes?